app/scripts/lib/account-tracker.js

```diff
@@ -83,6 +83,9 @@ export default class AccountTracker {
     this.preferencesController = opts.preferencesController;
     this.onboardingController = opts.onboardingController;
 
+    // subscribe to account removal
+    opts.onAccountRemoved((address) => this.removeAccount([address]));
```

Comment:
It'd be great to have a test that shows this is working

Reply:
I just did add some tests @Gudahtt